packages/target-decisioning-engine/src/artifactProvider.spec.js

```diff
@@ -280,3 +287,67 @@ describe("artifactProvider", () => {
     });
   });
 });
+
+describe("determineArtifactLocation", () => {
+  it("CDN host environment can be overridden", () => {
+    expect(
+      determineArtifactLocation({
+        client: "someClientId",
+        cdnEnvironment: "staging"
+      })
+    ).toEqual(
+      `${CDN_BASE_STAGE}/someClientId/production/v${SUPPORTED_ARTIFACT_MAJOR_VERSION}/rules.json`
```

Comment:
We might need a test for version which is an empty space, like the following:

```
.../{client}/{env}/rules.json
```

Reply:
> We might need a test for version which is an empty space, like the following:
> 
> ```
> .../{client}/{env}/rules.json
> ```

Actually, starting with this version of the SDK, a version is always required in the URL.  The reason we keep publishing artifacts without version is to support prior releases of the alpha.